detox/jest.startup.js

```diff
@@ -17,6 +17,9 @@ beforeAll(async () => {
 }, 120000);
 
 beforeEach(async () => {
+    await waitFor(element(by.id("NativeHome.Widgets")))
```

Comment:
I think this beforeEach isn't necessary, because you already wait for the specific buttons in the specs.